src/wasm.h

```diff
@@ -149,7 +149,8 @@ struct Literal {
     double f64;
   };
 
-  Literal() : type(WasmType::none), f64(0) {}
+  Literal() : Literal(WasmType::none) {}
+  explicit Literal(WasmType type) : type(type) { memset(&f64, 0, sizeof(f64)); }
```

Comment:
I made all of them explicit to avoid these types of ambiguity issues. It should be working as expected now? We could also use `enum class` to make things even more explicit.